modyn/selector/internal/selector_strategies/abstract_selection_strategy.py

```diff
@@ -266,6 +278,8 @@ def trigger(self) -> tuple[int, int, int, dict[str, Any]]:
         Returns:
             tuple[int, int, int]: Trigger ID, how many keys are in the trigger, number of overall partitions
         """
+        self._update_next_trigger_id()
+
```

Comment:
An sync at this place might not be early enough, as we overwrite `trigger()` in subclasses
https://github.com/eth-easl/modyn/blob/607dffffa673b42101ec0977efbc9aef0f172fb4/modyn/selector/internal/selector_strategies/coreset_strategy.py#L67

Reply:
rip. ok i need to rethink how we do this then

Reply:
tried to fix it at the selector side. actually we also need to update it before `inform_data`

Reply:
this should work!